haystack/dataclasses/byte_stream.py

```diff
@@ -24,21 +25,36 @@ def to_file(self, destination_path: Path):
 
     @classmethod
     def from_file_path(
-        cls, filepath: Path, mime_type: Optional[str] = None, meta: Optional[Dict[str, Any]] = None
+        cls,
+        filepath: Path,
+        mime_type: Optional[str] = None,
+        meta: Optional[Dict[str, Any]] = None,
+        mime_type_resolution_priority: Optional[List[str]] = None,
```

Comment:
don't we need a default here? Otherwise the code should crash, right? because the `None` here will override your `default_factory` and 💥

Reply:
But None is not passed, it has an or guard on the loc 47

Reply:
ah, I see. All good